<launch>
  <!-- Base Args -->
  <arg name="sim" default="false" doc="Whether to run in Aikido-sim only" />
  <arg name="version" default="2" doc="Kinova Arm Version (JACO = 2, Kortex = 3)" />
  <arg name="use_forque" default="false" />

  <!-- Locating Args -->
  <arg name="autoRestart" default="false" doc="Whether to restart behavior tree at end without command line input." />
  <arg name="treeFile" default="full_plate_locator.xml" doc="BT root XML, from package://ada_feeding/trees" />

  <!-- Libada Config -->
  <include file="$(find libada)/launch/libada.launch" pass_all_args="true" />
  <!-- Libada Overrides -->
  <rosparam file="$(find ada_feeding)/config/gen$(arg version)_ada_config.yaml" command="load" ns="adaConf" />

  <!-- If No Forque, revert to basic traj controller -->
  <group unless="$(arg use_forque)">
    <rosparam param="adaConf/arm_controller">trajectory_controller</rosparam>
  </group>


  <!-- Launch Locating Binary -->
  <node name="locating"
          pkg="ada_feeding"
          type="locating"
          output="screen"
          required="true">
    <param name="sim" value="$(arg sim)" />
    <param name="use_forque" value="$(arg use_forque)" />
    <param name="autoRestart" value="$(arg autoRestart)" />
    <param name="treeFile" value="$(find ada_feeding)/trees/$(arg treeFile)" />
    <rosparam file="$(find ada_feeding)/config/gen$(arg version)_feeding_config.yaml" command="load" />
  </node>

</launch>